core/app/components/pubs/PubEditor/actions.ts

```diff
@@ -114,16 +120,38 @@ export const updatePub = defineServerAction(async function updatePub({
 	});
 
 	try {
-		const result = await _updatePub({
-			pubId,
+		const updateQuery = PubOp.update(pubId, {
 			communityId: community.id,
-			pubValues,
-			stageId,
-			continueOnValidationError,
 			lastModifiedBy,
+			continueOnValidationError,
 		});
 
-		return result;
+		if (stageId) {
+			updateQuery.setStage(stageId);
+		}
+
+		const { values: processedVals }: { values: typeof pubValues } =
+			parseRichTextForPubFieldsAndRelatedPubs({
+				pubId: pubId,
+				values: pubValues as Record<string, JsonValue>,
+			});
```

Comment:
it would probably make sense to put this call in pub-op.ts somewhere, but we'd need to make it work with normalized pubValues first

Reply:
oh good point, we're indeed not doing that. should not be hard to add